src/net/wooga/jenkins/pipeline/BuildUtils.groovy

```diff
@@ -0,0 +1,66 @@
+package net.wooga.jenkins.pipeline
+
+import org.jenkinsci.plugins.pipeline.modeldefinition.Utils
+
+class BuildVersion {
+    final String version
+    final Boolean optional
+
+    BuildVersion(version, optional) {
+        this.version = version
+        this.optional = optional
+    }
+    
+    @Override
+    String toString() {
+        return toString(false).toString()
+    }
+
+    String toString(Boolean detailed)
+    {
+        if (detailed){
+            return "version : ${version}, optional : ${optional}"
+        }
+        return version
+    }
+}
+
+// For each entry in a versions map, if they are not in the standard format
+// we process them into the newer format that supports optional arguments per version
+// (such as if it's optional, etc)
+// Example: '3001.x' -> [ version : '3001.x', optional : false ]
```

Comment:
You should look into javadocs/groovy doc strings.
They work similar to what we have in c#.